pkg/scheduler/queue/user_queues.go

```diff
@@ -28,298 +27,350 @@ type querier struct {
 	disconnectedAt time.Time
 }
 
-// This struct holds user queues for pending requests. It also keeps track of connected queriers,
-// and mapping between users and queriers.
-type queues struct {
-	userQueues map[string]*userQueue
-
-	// List of all users with queues, used for iteration when searching for next queue to handle.
-	// Users removed from the middle are replaced with "". To avoid skipping users during iteration, we only shrink
-	// this list when there are ""'s at the end of it.
-	users []string
-
-	maxUserQueueSize int
+type tenantQuerierState struct {
```

Comment:
Maybe a name like `tenantQuerierAssignments` would make the purpose of this type clearer? If I understand your intention correctly, the idea is that this type owns the assignment of tenants to queriers based on shuffle sharding.

Reply:
done